### PR TITLE
Move in-package resilience bypassing logic to a resilience check with accessing module

### PR DESCRIPTION
**Pull request by elsh**
isResilient() returns the decl's resilience by definition and should not be altered whether resilience bypassing optimization is enabled or not. The overloaded isResilient(ModuleDecl, ResilienceExpansion) is used for accessing a decl from a client module, which is the appropriate place to check for in-package resilience bypass optimization.

Resolves rdar://123031292

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -4206,6 +4205,14 @@ bool ValueDecl::hasOpenAccess(const DeclContext *useDC) const {
   return access == AccessLevel::Open;
 }
 
+bool ValueDecl::bypassResilienceInPackage(ModuleDecl *accessingModule) const {
+  return getASTContext().LangOpts.EnableBypassResilienceInPackage &&
+          getModuleContext()->inSamePackage(accessingModule) &&
+          !getModuleContext()->isBuiltFromInterface() &&
+          getFormalAccessScope(/*useDC=*/nullptr,
+                               /*treatUsableFromInlineAsPublic=*/true).isPackage();
```
Comment (slavapestov) [inline lib/AST/Decl.cpp:4213]: I think you don't need the isPackage() check -- it should be safe to access public decls non-resiliently too across a package boundary.

**Review by slavapestov — APPROVED** [review]
Thank you!

**Review by CodaFi — APPROVED** [review]

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (elsh) [reply lib/AST/Decl.cpp:4213]: I'll push a separate PR for that.

**Pull request merged**